js/js_test.go

```diff
@@ -21,3 +22,87 @@ func TestEvaluateRaceCondition(t *testing.T) {
 		_, _ = Evaluate("true", params, nil)
 	}()
 }
+
+func TestTimeFunctions(t *testing.T) {
+	timestamp := int64(7323)
+	t.Run("failing hour() function", func(t *testing.T) {
+		entity := corev2.FixtureEvent("foo", "bar")
+		entity.Timestamp = timestamp
+		synth := dynamic.Synthesize(entity)
+		params := map[string]interface{}{"event": synth}
+
+		result, err := Evaluate("hour(event.timestamp) != 2", params, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, false, result)
+	})
+	t.Run("failing minute() function", func(t *testing.T) {
+		entity := corev2.FixtureEvent("foo", "bar")
+		entity.Timestamp = timestamp
+		synth := dynamic.Synthesize(entity)
+		params := map[string]interface{}{"event": synth}
+
+		result, err := Evaluate("minute(event.timestamp) != 2", params, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, false, result)
+	})
+	t.Run("failing second() function", func(t *testing.T) {
+		entity := corev2.FixtureEvent("foo", "bar")
+		entity.Timestamp = timestamp
+		synth := dynamic.Synthesize(entity)
+		params := map[string]interface{}{"event": synth}
+
+		result, err := Evaluate("second(event.timestamp) != 3", params, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, false, result)
+	})
+	t.Run("failing seconds_since() function", func(t *testing.T) {
+		entity := corev2.FixtureEvent("foo", "bar")
+		entity.Timestamp = timestamp
+		synth := dynamic.Synthesize(entity)
+		params := map[string]interface{}{"event": synth}
+
+		result, err := Evaluate("seconds_since(event.timestamp) <= 10000", params, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, false, result)
+	})
+	t.Run("passing hour() function", func(t *testing.T) {
+		entity := corev2.FixtureEvent("foo", "bar")
+		entity.Timestamp = timestamp
+		synth := dynamic.Synthesize(entity)
+		params := map[string]interface{}{"event": synth}
+
+		result, err := Evaluate("hour(event.timestamp) == 2", params, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, true, result)
+	})
+	t.Run("passing minute() function", func(t *testing.T) {
+		entity := corev2.FixtureEvent("foo", "bar")
+		entity.Timestamp = timestamp
+		synth := dynamic.Synthesize(entity)
+		params := map[string]interface{}{"event": synth}
+
+		result, err := Evaluate("minute(event.timestamp) == 2", params, nil)
```

Comment:
I don't really understand this use case. The timestamp is going to be a unix time stamp... seconds since the epoch. Is this how the function is intended to be used?